qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1092,38 +1092,81 @@ def update(self, md_template):
 
             # simple validations of sample ids and column names
             samples_diff = set(new_map.index).difference(current_map.index)
+            # Fail if we don't have the same samples
             if samples_diff:
                 raise QiitaDBError(
                     'The new template differs from what is stored '
                     'in database by these samples names: %s'
                     % ', '.join(samples_diff))
-            columns_diff = set(new_map.columns).difference(current_map.columns)
-            if columns_diff:
+
+            # Fail if we have new columns in the templates
+            # (only updates are allowed, not extensions)
+            if not set(current_map.columns).issuperset(new_map.columns):
+                columns_diff = set(new_map.columns).difference(
+                    current_map.columns)
                 raise QiitaDBError(
                     'The new template differs from what is stored '
                     'in database by these columns names: %s'
                     % ', '.join(columns_diff))
 
-            # here we are comparing two dataframes following:
-            # http://stackoverflow.com/a/17095620/4228285
-            current_map.sort(axis=0, inplace=True)
-            current_map.sort(axis=1, inplace=True)
-            new_map.sort(axis=0, inplace=True)
-            new_map.sort(axis=1, inplace=True)
-            map_diff = (current_map != new_map).stack()
-            map_diff = map_diff[map_diff]
-            map_diff.index.names = ['id', 'column']
-            changed_cols = map_diff.index.get_level_values('column').unique()
-
-            if not self.can_be_updated(columns=set(changed_cols)):
+            # In order to speed up some computation, let's compare only the
+            # common columns. current_map.columns is a superset of
+            # new_map.columns, so this will not fail
+            current_map = current_map[new_map.columns]
+
+            # Get the values that we need to change
+            diff_map = current_map != new_map
+            ne_stacked = diff_map.stack()
+            changed = ne_stacked[ne_stacked]
```

Comment:
See pandas boolean indexing (changed only holds the values that ne_stacked had a True)